Engine/Helper.cs

```diff
@@ -3555,4 +3555,239 @@ public object VisitUsingExpression(UsingExpressionAst usingExpressionAst)
             return null;
         }
     }
+
+    /// Class to represent a directed graph
+    public class Digraph<T>
```

Comment:
Consider making this an internal class for now. We can open it up as public as the scenarios evolve.